### PR TITLE
fix: incorrect topup

### LINKED ISSUE #3136: TopUp amount is not credited / incorrectly credited to stamps
⚠️ Support requests in an issue-format will be closed immediately. For support, go to Swarm's [Discord](https://discord.gg/XGNBEKktkD).

### Summary

TopUp amount is not credited to stamp because the functionality does not exist in **HandleTopUp func** in **mock/service**. TopUp amount is also calculated incorrectly from normalized balance. 
### Expected behavior

**/stamps/topup{id}/{amount}** should add the credited amount to stamp

### Actual behavior

Topup is not credited to stamp as seen in the screenshot below. Topup should add the amount to the specific stamp but the amount does not change.

If a fix is applied, we get another bug, that is incorrect amount which is added to the amount and can be seen after api call. This incorrect amount is displayed because of the normalized balance received from smart contract which needs to be divided by 2^depth to get the actual amount.
This behavior was also verified by @istae  and @acud in production code. 

### Steps to reproduce

Please use the following commands on master branch to reproduce the issue.
`curl localhost:1635/stamps`
`curl -XPOST localhost:1635/stamps/1000/20`
`curl -XPATCH localhost:1635/stamps/topup/{stampID}/1000`
`curl localhost:1635/stamps`

To verify incorrect topup, you can modify and add the HandleTopUp function in mock/service with 
```
        for _, v := range m.issuersMap {
		if bytes.Equal(batchID, v.ID()) {
			if newBalance.Cmp(v.Amount()) > 0 {
				v.Amount().Set(newBalance)
			}
			return
		}
	}
```

### Possible solution
A possible solution for this issue is to correctly calculate the topup amount from normalized balance using **normalizedBalance / 2^Depth** and adding the functionality in HandleTopUp in mock/service.

### PR DESCRIPTION
**Pull request by swarmHaseeb**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md)
- [x] My change requires a documentation update and I have done it
- [x] I have added tests to cover my changes.

### Related Issue (Optional)

This PR fixes #3136 
### Screenshots (if appropriate):

---
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3130)

### CODE REVIEW

**Review by mrekucci — COMMENTED** [review]
The commit message should use the imperative mood; `fix: incorrect topup` should do.

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3130#-:-N83cgWB6hrA-ycTxRaf:b-iclcbd)* status: all files reviewed, 1 unresolved discussion (waiting on @swarmHaseeb)

___
*[`pkg/node/devnode.go` line 275 at r1](https://reviewable.io/reviews/ethersphere/bee/3130#-N83cGBP4Yt-FWFoGXrU:-N83cGBP4Yt-FWFoGXrV:bbvkteb) ([raw file](https://github.com/ethersphere/bee/blob/1b0324bbbefdcffa01dee0fe5aa7abe3d941a619/pkg/node/devnode.go#L275)):*
> ```Go
> 					return err
> 				}
> 				_totalAmount := big.NewInt(newBalance.Int64() / int64(math.Pow(2, float64(batch.Depth))))
> ```

Why the underscore prefix?

**Review by swarmHaseeb — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3130)* status: all files reviewed, 1 unresolved discussion (waiting on @mrekucci)

___
*[`pkg/node/devnode.go` line 275 at r1](https://reviewable.io/reviews/ethersphere/bee/3130#-N83cGBP4Yt-FWFoGXrU:-N83dWOAAvfdJLQfzaCT:b-wk0aiw) ([raw file](https://github.com/ethersphere/bee/blob/1b0324bbbefdcffa01dee0fe5aa7abe3d941a619/pkg/node/devnode.go#L275)):*
<details><summary><i>Previously, mrekucci wrote…</i></summary><blockquote>

Why the underscore prefix?
</blockquote></details>

ummm totalAmount, topupAmount and newBalance was already taken and total amount made more sense here and i didnt wanted to override the other values in case someone wanted to double check and verify values for debugging .

**Review by swarmHaseeb — COMMENTED** [review]
edited.

> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3130#-:-N83eW1n4g_wwW5pv-eP:b4bani8)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @swarmHaseeb)

**Review by mrekucci — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3130)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @acud and @istae)

___
*[`pkg/node/devnode.go` line 275 at r1](https://reviewable.io/reviews/ethersphere/bee/3130#-N83cGBP4Yt-FWFoGXrU:-N83gksF7kTSYeX4IN7N:b-8pzwl1) ([raw file](https://github.com/ethersphere/bee/blob/1b0324bbbefdcffa01dee0fe5aa7abe3d941a619/pkg/node/devnode.go#L275)):*
<details><summary><i>Previously, swarmHaseeb (Haseeb Raja) wrote…</i></summary><blockquote>

ummm totalAmount, topupAmount and newBalance was already taken and total amount made more sense here and i didnt wanted to override the other values in case someone wanted to double check and verify values for debugging .
</blockquote></details>

You can name it `topUpTotalAmount` (or something similar) if you're concerned about debugging, but [avoid constructs with `_` prefix](https://go.dev/doc/effective_go#names), which doesn't add much value to readability.

**Comment by aloknerurkar** [comment]
@istae Is there an issue for this?

I dont think this is the right usage of TopUp. TopUp is the topup amount per chunk. This is not the new value of the batch. The normalized balance is not calculated here. The variable name might be misleading.

File: pkg/postage/batchservice/batchservice.go [hunk 1]
```diff
@@ -146,8 +147,10 @@ func (svc *batchService) TopUp(id []byte, normalisedBalance *big.Int, txHash []b
 		return fmt.Errorf("update: %w", err)
 	}
 
+	topUpAmount := big.NewInt(normalisedBalance.Int64() / int64(math.Pow(2, float64(b.Depth))))
```
Comment (istae) [inline pkg/postage/batchservice/batchservice.go:150]: please use big int division, something like 
`topUpAmount := new(big.Int).Div(normalisedBalance, 1 << b.depth)`

File: pkg/node/devnode.go [hunk 2]
```diff
@@ -271,8 +272,8 @@ func NewDevBee(logger logging.Logger, o *DevOptions) (b *DevBee, err error) {
 				if err != nil {
 					return err
 				}
-
-				post.HandleTopUp(batch.ID, newBalance)
+				topUptotalAmount := big.NewInt(newBalance.Int64() / int64(math.Pow(2, float64(batch.Depth))))
```
Comment (istae) [inline pkg/node/devnode.go:275]: please use big int div

**Review by istae — CHANGES_REQUESTED** [review]
great, just a few comments

**Comment by istae** [comment]
> @istae Is there an issue for this?
> 
> I dont think this is the right usage of TopUp. TopUp is the topup amount per chunk. This is not the new value of the batch. The normalized balance is not calculated here. The variable name might be misleading.

@aloknerurkar so we caught this bug with @swarmHaseeb this morning where when you top up a batch, the amount field is replaced with the value of the batch. There is no issue, maybe we should create one for bookkeeping purposes.

File: pkg/postage/batchservice/batchservice.go (hunk 1 above)
Comment (aloknerurkar) [reply pkg/postage/batchservice/batchservice.go:150]: This is called normalisedBalance but it is the topup amount per chunk. If you divide it by no. of chunks what would this value be?

Comment (istae) [reply pkg/postage/batchservice/batchservice.go:150]: @aloknerurkar is right, @swarmHaseeb you missed the change we discussed which was passing `TopUpAmount` to `TopUp` from inside postage/listener.go line 140

Comment (istae) [reply pkg/postage/batchservice/batchservice.go:150]: [it should be new(big.Int).Div(topUpAmount, 1 << b.depth)](https://github.com/ethersphere/storage-incentives/blob/master/src/PostageStamp.sol#L151)

**Review by istae — CHANGES_REQUESTED** [review]

**Comment by swarmHaseeb** [comment]
> > @istae Is there an issue for this?
> > I dont think this is the right usage of TopUp. TopUp is the topup amount per chunk. This is not the new value of the batch. The normalized balance is not calculated here. The variable name might be misleading.
> 
> @aloknerurkar so we caught this bug with @swarmHaseeb this morning where when you top up a batch, the amount field is replaced with the value of the batch. There is no issue, maybe we should create one for bookkeeping purposes.

Please find the issue [here](https://app.zenhub.com/workspaces/bee-active-sprint-6183a7dddf3a6e001639c89e/issues/ethersphere/bee/3136)

**Review by swarmHaseeb — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3130)* status: 0 of 10 files reviewed, 3 unresolved discussions (waiting on @acud, @istae, and @mrekucci)

___
*[`pkg/node/devnode.go` line 275 at r2](https://reviewable.io/reviews/ethersphere/bee/3130#-N83a_GUsGmfNYidNu0z-r2-275:-N8NgDIa-coTyM5kB3mq:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/c8145800ea9e32b1a790832219ab9eb7905ec45a/pkg/node/devnode.go#L275)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

please use big int div
</blockquote></details>

Done.
___
*[`pkg/postage/batchservice/batchservice.go` line 150 at r2](https://reviewable.io/reviews/ethersphere/bee/3130#-N83a_GUsGmfNYidNu1--r2-150:-N8YCuP_2dKmXqRJSP13:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/c8145800ea9e32b1a790832219ab9eb7905ec45a/pkg/postage/batchservice/batchservice.go#L150)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

[it should be new(big.Int).Div(topUpAmount, 1 << b.depth)](https://github.com/ethersphere/storage-incentives/blob/master/src/PostageStamp.sol#L151)
</blockquote></details>

Done.
___
*[`pkg/postage/mock/service.go` line 95 at r2](https://reviewable.io/reviews/ethersphere/bee/3130#-N83a_GUsGmfNYidNu10-r2-95:-N8YCxZ3-GqZCTzkfKsi:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/c8145800ea9e32b1a790832219ab9eb7905ec45a/pkg/postage/mock/service.go#L95)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

arg name should be `amount`
</blockquote></details>

Done.

File: .github/workflows/beekeeper.yml [hunk 3]
```diff
@@ -14,7 +14,7 @@ jobs:
       REPLICA: 3
       RUN_TYPE: "PR RUN"
       SETUP_CONTRACT_IMAGE_TAG: "0.2.0"
-      BEEKEEPER_BRANCH: "master"
+      BEEKEEPER_BRANCH: "topup-bug"
```
Comment (istae) [inline .github/workflows/beekeeper.yml:17]: If this branch fixes the beekeeper issue, the next steps are :
1. merge the beekeeper branch to master
2. change this line back to master

File: pkg/postage/mock/service.go [hunk 4]
```diff
@@ -89,9 +90,21 @@ func (m *mockPostage) IssuerUsable(_ *postage.StampIssuer) bool {
...
 
-func (m *mockPostage) HandleTopUp(_ []byte, _ *big.Int) {}
+func (m *mockPostage) HandleTopUp(batchID []byte, amount *big.Int) {
+	m.issuerLock.Lock()
+	defer m.issuerLock.Unlock()
+
+	for _, v := range m.issuersMap {
+		if bytes.Equal(batchID, v.ID()) {
+			if amount.Cmp(v.Amount()) > 0 {
+				v.Amount().Set(amount)
```
Comment (istae) [inline pkg/postage/mock/service.go:102]: this should be an addition, not `Set`

File: pkg/postage/service.go [hunk 5]
```diff
@@ -116,29 +116,29 @@ func (ps *service) HandleCreate(b *Batch) error {
...
+func (ps *service) HandleDepthIncrease(batchID []byte, newDepth uint8, newBalance *big.Int) {
 	ps.lock.Lock()
 	defer ps.lock.Unlock()
 
 	for _, v := range ps.issuers {
 		if bytes.Equal(batchID, v.data.BatchID) {
 			if newDepth > v.data.BatchDepth {
 				v.data.BatchDepth = newDepth
-				v.data.BatchAmount = normalisedBalance
+				v.data.BatchAmount = newBalance
```
Comment (istae) [inline pkg/postage/service.go:141]: `BatchAmount` is not the balance!

**Review by istae — CHANGES_REQUESTED** [review]

**Review by swarmHaseeb — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3130)* status: 0 of 10 files reviewed, 6 unresolved discussions (waiting on @acud, @istae, and @mrekucci)

___
*[`.github/workflows/beekeeper.yml` line 17 at r7](https://reviewable.io/reviews/ethersphere/bee/3130#-N8Tg24V5BveXKm45Kjd-r7-17:-N8YPkbp8wLd927F9qbm:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/5e36e1356b3f9e58be1b6edbbb44e63666c2dda9/.github/workflows/beekeeper.yml#L17)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

If this branch fixes the beekeeper issue, the next steps are :
1. merge the beekeeper branch to master
2. change this line back to master
</blockquote></details>

Done.
___
*[`pkg/postage/service.go` line 141 at r7](https://reviewable.io/reviews/ethersphere/bee/3130#-N8O9fvF9SK2mohV4cKU-r7-141:-N8YPhsT-qzg_t__DYka:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/5e36e1356b3f9e58be1b6edbbb44e63666c2dda9/pkg/postage/service.go#L141)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

`BatchAmount` is not the balance
</blockquote></details>

Done.
___
*[`pkg/postage/mock/service.go` line 102 at r7](https://reviewable.io/reviews/ethersphere/bee/3130#-N83a_GUsGmfNYidNu10-r7-102:-N8YPjDv68H2LvGd6LSc:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/5e36e1356b3f9e58be1b6edbbb44e63666c2dda9/pkg/postage/mock/service.go#L102)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

this should be an addition, not `Set`
</blockquote></details>

Done.

File: pkg/postage/service.go [hunk 6]
```diff
@@ -116,29 +116,29 @@ func (ps *service) HandleCreate(b *Batch) error {
...
+func (ps *service) HandleDepthIncrease(batchID []byte, newDepth uint8, newBatchAmount *big.Int) {
 	ps.lock.Lock()
 	defer ps.lock.Unlock()
 
 	for _, v := range ps.issuers {
 		if bytes.Equal(batchID, v.data.BatchID) {
 			if newDepth > v.data.BatchDepth {
 				v.data.BatchDepth = newDepth
-				v.data.BatchAmount = normalisedBalance
+				v.data.BatchAmount = newBatchAmount
```
Comment (istae) [inline pkg/postage/service.go:141]: `HandleDepthIncrease` gets passed the balance, not the amount. You cannot update the amount with balance. 
You can remove the balance argument.

File: .github/workflows/beekeeper.yml [hunk 7]
```diff
@@ -140,7 +140,7 @@ jobs:
...
```
Comment (istae) [inline .github/workflows/beekeeper.yml:143]: replace with master

**Review by istae — CHANGES_REQUESTED** [review]

File: .github/workflows/beekeeper.yml [hunk 8]
```diff
@@ -230,7 +230,7 @@ jobs:
...
```
Comment (istae) [inline .github/workflows/beekeeper.yml:233]: also here

**Review by istae — CHANGES_REQUESTED** [review]

File: pkg/postage/mock/service.go [hunk 9]
```diff
@@ -89,11 +90,23 @@ func (m *mockPostage) IssuerUsable(_ *postage.StampIssuer) bool {
...
+func (m *mockPostage) HandleCreate(_ *postage.Batch, _ *big.Int) error { return nil }
 
-func (m *mockPostage) HandleTopUp(_ []byte, _ *big.Int) {}
+func (m *mockPostage) HandleTopUp(batchID []byte, amount *big.Int) {
+	m.issuerLock.Lock()
+	defer m.issuerLock.Unlock()
+
+	for _, v := range m.issuersMap {
+		if bytes.Equal(batchID, v.ID()) {
+			if amount.Cmp(v.Amount()) > 0 {
```
Comment (istae) [inline pkg/postage/mock/service.go:101]: this check is unnecessary. remove please

File: pkg/postage/service.go [hunk 10]
```diff
@@ -116,29 +116,28 @@ func (ps *service) HandleCreate(b *Batch) error {
...
-func (ps *service) HandleTopUp(batchID []byte, newValue *big.Int) {
+func (ps *service) HandleTopUp(batchID []byte, amount *big.Int) {
 	ps.lock.Lock()
 	defer ps.lock.Unlock()
 
 	for _, v := range ps.issuers {
 		if bytes.Equal(batchID, v.data.BatchID) {
-			if newValue.Cmp(v.data.BatchAmount) > 0 {
-				v.data.BatchAmount = newValue
+			if amount.Cmp(v.data.BatchAmount) > 0 {
```
Comment (istae) [inline pkg/postage/service.go:125]: remove check

**Review by istae — CHANGES_REQUESTED** [review]

**Review by swarmHaseeb — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3130)* status: 0 of 10 files reviewed, 11 unresolved discussions (waiting on @acud, @istae, and @mrekucci)

___
*[`pkg/postage/service.go` line 141 at r8](https://reviewable.io/reviews/ethersphere/bee/3130#-N8O9fvF9SK2mohV4cKU-r8-141:-N8btbMG9tTwY94RT_lC:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/66077cb24002394106ba8ca857f31c0f2f7bf98e/pkg/postage/service.go#L141)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

`HandleDepthIncrease` gets passed the balance, not the amount. You cannot update the amount with balance. 
You can remove the balance argument. 
</blockquote></details>

Done.
___
*[`pkg/postage/service.go` line 125 at r9](https://reviewable.io/reviews/ethersphere/bee/3130#-N8O9fvF9SK2mohV4cKU-r9-125:-N8bt_l5-1hPNWNqF_Vd:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/327b3c9b65bc997a936da428da75308ee03799a5/pkg/postage/service.go#L125)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

remove check
</blockquote></details>

Done.
___
*[`pkg/postage/mock/service.go` line 101 at r9](https://reviewable.io/reviews/ethersphere/bee/3130#-N83a_GUsGmfNYidNu10-r9-101:-N8buRMn0rG6hr82zHRM:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/327b3c9b65bc997a936da428da75308ee03799a5/pkg/postage/mock/service.go#L101)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

this check is unnecessary. remove please
</blockquote></details>

Done.
___
*[`.github/workflows/beekeeper.yml` line 143 at r8](https://reviewable.io/reviews/ethersphere/bee/3130#-N8Tg24V5BveXKm45Kjd-r8-143:-N8btQcbDoZEmX3LBuET:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/66077cb24002394106ba8ca857f31c0f2f7bf98e/.github/workflows/beekeeper.yml#L143)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

replace with master
</blockquote></details>

Done.
___
*[`.github/workflows/beekeeper.yml` line 233 at r8](https://reviewable.io/reviews/ethersphere/bee/3130#-N8Tg24V5BveXKm45Kjd-r8-233:-N8btRigDmOYVUwxdI4b:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/66077cb24002394106ba8ca857f31c0f2f7bf98e/.github/workflows/beekeeper.yml#L233)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

also here
</blockquote></details>

Done.

File: pkg/node/devnode.go [hunk 11]
```diff
@@ -271,8 +271,9 @@ func NewDevBee(logger logging.Logger, o *DevOptions) (b *DevBee, err error) {
 				if err != nil {
 					return err
 				}
+				topUpTotalAmount := big.NewInt(0).Div(batch.Value, big.NewInt(int64(1<<(batch.Depth))))
 
-				post.HandleTopUp(batch.ID, newBalance)
+				post.HandleTopUp(batch.ID, topUpTotalAmount)
```
Comment (istae) [inline pkg/node/devnode.go:276]: should just be `topupAmount`

File: pkg/postage/batchservice/batchservice_test.go [hunk 12]
```diff
@@ -120,6 +120,7 @@ func TestBatchServiceCreate(t *testing.T) {
 			testBatch.ID,
 			testBatch.Owner,
 			testBatch.Value,
+			testBatch.Value,
```
Comment (istae) [inline pkg/postage/batchservice/batchservice_test.go:123]: repeating `testBatch.Value` ?

File: pkg/postage/batchservice/batchservice_test.go [hunk 13]
```diff
@@ -146,6 +147,7 @@ func TestBatchServiceCreate(t *testing.T) {
...
```
Comment (istae) [inline pkg/postage/batchservice/batchservice_test.go:150]: here as well

File: pkg/postage/batchservice/batchservice_test.go [hunk 14]
```diff
@@ -180,6 +182,7 @@ func TestBatchServiceCreate(t *testing.T) {
...
```
Comment (istae) [inline pkg/postage/batchservice/batchservice_test.go:185]: here again

**Review by istae — CHANGES_REQUESTED** [review]
Great work so far!

Some unit tests need to be fixed.

**Review by swarmHaseeb — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3130)* status: 0 of 10 files reviewed, 15 unresolved discussions (waiting on @acud, @aloknerurkar, @istae, and @mrekucci)

___
*[`pkg/node/devnode.go` line 276 at r10](https://reviewable.io/reviews/ethersphere/bee/3130#-N83a_GUsGmfNYidNu0z-r10-276:-N8w_n3JDYw4xfiHlaDZ:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/2f9d1a4b96f187942f13c3adab455929d1080877/pkg/node/devnode.go#L276)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

should just be `topupAmount`
</blockquote></details>

Done.
___
*[`pkg/postage/batchservice/batchservice_test.go` line 123 at r10](https://reviewable.io/reviews/ethersphere/bee/3130#-N8TJyw52Uy5lDivjnsY-r10-123:-N8we1t7CajuP--MFonX:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/2f9d1a4b96f187942f13c3adab455929d1080877/pkg/postage/batchservice/batchservice_test.go#L123)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

repeating `testBatch.Value` ?
</blockquote></details>

Done.
___
*[`pkg/postage/batchservice/batchservice_test.go` line 150 at r10](https://reviewable.io/reviews/ethersphere/bee/3130#-N8TJyw52Uy5lDivjnsY-r10-150:-N8we2e28lnH5PFdxA2W:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/2f9d1a4b96f187942f13c3adab455929d1080877/pkg/postage/batchservice/batchservice_test.go#L150)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

here as well 
</blockquote></details>

Done.
___
*[`pkg/postage/batchservice/batchservice_test.go` line 185 at r10](https://reviewable.io/reviews/ethersphere/bee/3130#-N8TJyw52Uy5lDivjnsY-r10-185:-N8we3Jd3dXi-YaydpZ6:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/2f9d1a4b96f187942f13c3adab455929d1080877/pkg/postage/batchservice/batchservice_test.go#L185)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

here again
</blockquote></details>

Done.

**Review by mrekucci — APPROVED** [review]
> Reviewed 1 of 3 files at r4, 1 of 6 files at r5, 3 of 8 files at r9, 3 of 3 files at r10, 2 of 2 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3130)* status: all files reviewed, 15 unresolved discussions (waiting on @acud, @aloknerurkar, and @istae)

___
*[`pkg/node/devnode.go` line 274 at r11](https://reviewable.io/reviews/ethersphere/bee/3130#-N95XXEv9ec7HDWMrpLC:-N95XXEv9ec7HDWMrpLD:b5p5y5i) ([raw file](https://github.com/ethersphere/bee/blob/7ef8011fdd24cc29373604c59f89d8123a102aac/pkg/node/devnode.go#L274)):*
> ```Go
> 					return err
> 				}
> 				topUpAmount := big.NewInt(0).Div(batch.Value, big.NewInt(int64(1<<(batch.Depth))))
> ```

The casting isn't needed, this will do: `1<<batch.Depth`.



_Code quote:_
```Go
big.NewInt(int64(1<<(batch.Depth))))
```
___
*[`pkg/postage/batchservice/batchservice.go` line 121 at r11](https://reviewable.io/reviews/ethersphere/bee/3130#-N95YXRa7tuabgPE9cgz:-N95YXRa7tuabgPE9ch-:b9y8h6g) ([raw file](https://github.com/ethersphere/bee/blob/7ef8011fdd24cc29373604c59f89d8123a102aac/pkg/postage/batchservice/batchservice.go#L121)):*
> ```Go
> 	}
> 
> 	amount := big.NewInt(0).Div(totalAmout, big.NewInt(int64(1<<(batch.Depth))))
> ```

The same as above: `1<<batch.Depth` will do.

**Review by notanatol — APPROVED** [review]

**Review by swarmHaseeb — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3130#-:-N97-5qF3CeTtA0DVOV2:bnfp4nl)* status: 4 of 10 files reviewed, 15 unresolved discussions (waiting on @acud, @aloknerurkar, @istae, and @mrekucci)

**Review by istae — APPROVED** [review]

**Pull request merged**